src/rebalancer/bridges/CCTPBridge.sol

```diff
@@ -0,0 +1,146 @@
+
+import {IBridge} from "src/interfaces/IBridge.sol";
+import {IRebalancer} from "src/interfaces/IRebalancer.sol";
+import {ImTokenMinimal} from "src/interfaces/ImToken.sol";
+
+import {BaseBridge} from "src/rebalancer/bridges/BaseBridge.sol";
+import {CCTPHelper} from "src/rebalancer/bridges/cctp/CCTPHelper.sol";
+
+contract CCTPBridge is BaseBridge, CCTPHelper, IBridge, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+
+    // ----------- STORAGE ------------
+    address public immutable rebalancer;
+    mapping(uint32 chainId => uint32 cctpDomain) public chainIdToDomain; 
+    
+    // ----------- EVENTS ------------
+    event Rebalanced(address indexed market, uint256 amount);
+    event DomainMappingUpdated(
+        uint32 indexed chainId,
+        uint32 indexed domain
+    );
+    event TokenAccepted(address indexed token, bool status);
+
+    // ----------- ERRORS ------------
+    error CCTPBridge_AddressNotValid();
+    error CCTPBridge_InvalidReceiver();
+    error CCTPBridge_TokenMismatch();
+    error CCTPBridge_NotImplemented();
+    error CCTPBridge_DomainNotSet();
+
+    constructor(
+        address _roles,
+        address _tokenMessenger,
+        address _messageTransmitter,
+        address _rebalancer
+    )
+        BaseBridge(_roles)
+        CCTPHelper(_tokenMessenger, _messageTransmitter)
+    {
+        require(_rebalancer != address(0), CCTPBridge_AddressNotValid());
+        rebalancer = _rebalancer;
+    }
+
+    // ----------- OWNER / CONFIG ------------
+    function setDomainMapping(uint32 chainId, uint32 domain)
+        external
+        onlyBridgeConfigurator
+    {
+        chainIdToDomain[chainId] = domain;
+        emit DomainMappingUpdated(chainId, domain);
+    }
+
+    function setAcceptedToken(address token, bool allowed)
+        external
+        onlyBridgeConfigurator
+    {
+        acceptedTokens[token] = allowed;
+        emit TokenAccepted(token, allowed);
+    }
+
+    // ----------- VIEW ------------
+    function getFee(
+        uint32,
+        bytes memory,
+        bytes memory
+    ) external pure override returns (uint256) {
+        // Fee automatically deducted
+        revert CCTPBridge_NotImplemented();
+    }
+
+    // ----------- SOURCE CHAIN ------------
+    function sendMsg(
+        uint256 _extractedAmount,
+        address _market,
+        uint32 _dstChainId,
+        address _token,
+        bytes memory,
+        bytes memory
+    ) external payable override onlyRebalancer {
+        require(_extractedAmount > 0, BaseBridge_AmountMismatch());
+
+        uint32 dstDomain = chainIdToDomain[_dstChainId];
+        uint32 srcDomain = chainIdToDomain[uint32(block.chainid)];
+
+        bytes memory payload = abi.encode(_market);
+
+        (CCTPMessage memory cctpMsg, ) = createAndBurn(
+            _token,
+            _extractedAmount,
+            dstDomain,
+            _toBytes32(address(this)),
```

Comment:
Unless the bridge is deployed to the exact same address on the destination chain, due to receiver being `address(this))`, Circle will mint to the wrong address and the destination bridge won’t be able to claim the funds. 

Consider a per-chain receiver mapping or explicit destination bridge address instead of assuming address parity.